src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Win32.cs

```diff
@@ -105,6 +105,31 @@ private static void PopulateAllSystemTimeZones(CachedData cachedData)
             }
         }
 
+        private static unsafe string? GetAlternativeId(string id)
+        {
+            char* buffer = stackalloc char[100];
+            if (!GlobalizationMode.Invariant && !GlobalizationMode.UseNls)
+            {
+                foreach (char c in id)
+                {
+                    // ICU uses some characters as a separator and trim the id at that character.
+                    // while we should fail if the Id contained one of these characters.
+                    if (c == '\\' || c == '\n' || c == '\r')
+                    {
+                        return null;
+                    }
+                }
+
+                int length = Interop.Globalization.IanaIdToWindowsId(id, buffer, 100);
```

Comment:
Can't we use a `ReadOnlySpan<char>` for the buffer and just return .ToString() on it?

Reply:
Good question, is this to avoid using pointers? I mean what the benefit to use ReadOnlySpan here?

Reply:
It would avoid using pointers and also would simplify the call to create a string right? I think overall is the same I was curious why you decided to use pointers rather than ROS.

Reply:
I didn't use ROS because I need to pass a pointer anyway to the pinvoke and if used ROS I'll need to slice it before creating the string. I can try it if you think it is worth it.

Reply:
It's fine as is, it was mostly curiosity.